code/modules/virus2/dishincubator.dm

```diff
@@ -58,34 +89,61 @@
 	mutatechance = initial(mutatechance) * max(1, scancount)
 	growthrate = initial(growthrate) + lasercount
 
-/obj/machinery/disease2/incubator/attackby(var/obj/B as obj, var/mob/user as mob)
+/obj/machinery/disease2/incubator/attackby(var/obj/I, var/mob/user)
 	. = ..()
+
+	if(stat & (BROKEN))
+		to_chat(user, "<span class='warning'>\The [src] is broken. Some components will have to be replaced before it can work again.</span>")
+		return FALSE
+
 	if(.)
 		return
-	if(!is_operational())
-		return FALSE
-	if(istype(B, /obj/item/weapon/reagent_containers/glass) || istype(B,/obj/item/weapon/reagent_containers/syringe))
-		if(beaker)
-			to_chat(user, "\A [beaker] is already loaded into the machine.")
-			return FALSE
-
-		if(user.drop_item(B, src))
-			beaker =  B
-			to_chat(user, "You add \the [B] to \the [src]!")
-			updateUsrDialog()
-			return TRUE
-	else
-		if(istype(B,/obj/item/weapon/virusdish))
-			if(dish)
-				to_chat(user, "A dish is already loaded into the machine.")
-				return FALSE
 
-			if(user.drop_item(B, src))
-				dish =  B
-				to_chat(user, "You add the dish to \the [src]!")
-				updateUsrDialog()
-				return TRUE
+	if (istype(I,/obj/item/weapon/virusdish))
+		addDish(I,user)
+		updateUsrDialog()
+		return TRUE
 
+/obj/machinery/disease2/incubator/proc/addDish(var/obj/item/weapon/virusdish/VD,var/mob/user,var/force_slot = 0)
+	if (VD.open)
+		if (!force_slot)
+			if (!dish1)
+				dish1 = VD
+			else if (!dish2)
+				dish2 = VD
+			else if (!dish3)
+				dish3 = VD
+			else
+				to_chat(user,"<span class='warning'>There is no more room inside \the [src]. Remove a dish first.</span>")
+				return null
+		else
+			switch (force_slot)
```

Comment:
I think making the dishes a list instead of 3 hardcoded slots would be better

Reply:
I've given it a shot but unlike for the centrifuge I'm using some specific defines to update the light overlays, so it's really not worth the effort.